frontend/src/components/common/Resource/Resource.tsx

```diff
@@ -894,3 +895,190 @@ export function ConditionsSection(props: { resource: KubeObjectInterface | null
+
+  const volumes = resource?.spec?.volumes;
+
+  if (!volumes) {
+    return null;
+  }
+
+  const namespace = resource?.metadata?.namespace;
+
+  /*
+   * printVolumeLink will print a working link that is set within the router using fields from the resource as params
+   */
+  function PrintVolumeLink(props: PrintVolumeLinkProps) {
+    const { volumeName, volumeKind, volume } = props;
+    const resourceClasses = ResourceClasses;
+    const classList = Object.keys(ResourceClasses);
+
+    for (const kind of classList) {
+      if (kind.toLowerCase() === volumeKind.toLowerCase()) {
+        const VolumeClass = resourceClasses[kind];
+        const volumeInstance = new VolumeClass(volume);
+        const volumeRoute = volumeInstance.detailsRoute;
+        const volumeNamespace = volumeInstance.isNamespaced ? namespace : null;
+        // we use the keys of the volume[volumeKind], which is the non class created volume item, to find the name of the volume source
+        const volumeKeys = Object.keys(volume[volumeKind]);
+        let detailName;
+
+        // where the volume[volumeKind] contains a "name" word in the keys, grab that value
+        for (const key of volumeKeys) {
+          if (key.includes('name')) {
+            detailName = volume[volumeKind][key];
+            // using if render fixes empty source field issue for items like emptyDir where name is = ''
+            if (detailName) {
+              return (
+                <Link
+                  routeName={volumeRoute}
+                  params={{ namespace: volumeNamespace, name: detailName }}
+                >
+                  {volumeName}
+                </Link>
+              );
+            }
+          }
+        }
+      }
+    }
+
+    return <Typography>{volumeName}</Typography>;
+  }
+
+  function volumeRows(volume: VolumeSectionProps['volume']) {
+    const currentVolume = volume;
+    if (!currentVolume) {
+      return [];
+    }
+
+    let volumeName = currentVolume.name;
+    let volumeKind: string = '';
+
+    const volumeKeys = Object.keys(currentVolume);
+
+    for (const key of volumeKeys) {
+      if (key === 'name') {
+        volumeName = currentVolume[key];
+      } else {
+        volumeKind = key;
+      }
+    }
+
+    function volumeDetailGrab(vol: VolumeSectionProps['volume']): VolumePrints {
+      // array for items that are printable
+      const directPrint = [];
+
+      // array for items that are not printable and need to be printed to yaml
+      const yamlPrint = [];
+
+      // loop over volumeKeys and check if the value is a string, number, or bool
+      for (const key in vol) {
+        if (!(vol[key] === '')) {
+          if (
+            typeof vol[key] === 'string' ||
+            typeof vol[key] === 'number' ||
+            typeof vol[key] === 'boolean'
+          ) {
+            directPrint.push({
+              volKey: key,
+              volValue: typeof vol[key] === 'boolean' ? JSON.stringify(vol[key]) : vol[key],
```

Comment:
I think that for converting the `vol[key]` to string, it's more correct to do `vol[key].toString()` instead of the `JSON.stringify`.